library/plugins/graph/graph-plugin/src/main/java/com/cdsap/talaiot/plugin/graph/GraphConfigurationProvider.kt

```diff
@@ -0,0 +1,30 @@
+package com.cdsap.talaiot.plugin.graph
+
+import com.cdsap.talaiot.logger.LogTrackerImpl
+import com.cdsap.talaiot.provider.PublisherConfigurationProvider
+import com.cdsap.talaiot.publisher.Publisher
+import com.cdsap.talaiot.publisher.graph.GraphPublisherFactoryImpl
+import com.cdsap.talaiot.publisher.graph.TaskDependencyGraphPublisher
+import org.gradle.api.Project
+import java.util.concurrent.Executors
+
+class GraphConfigurationProvider(
+    private val project: Project
+) : PublisherConfigurationProvider {
+    override fun get(): List<Publisher> {
+        val publishers = mutableListOf<Publisher>()
+        val talaiotExtension = project.extensions.getByName("talaiot") as GraphExtension
+
+        talaiotExtension.publishers?.apply {
+            publishers.add(
+                TaskDependencyGraphPublisher(
+                    this.taskDependencyGraphPublisher!!,
```

Comment:
The `!!` in here feels a bit dangerous. Since we are just moving the current logic around I'm leaving it as is, but it would be nice to add a check or even a safer way to get this publisher instance.

Reply:
https://github.com/cdsap/Talaiot/issues/231